// @react-website-themes/classy-docs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Article", {
  enumerable: true,
  get: function get() {
    return _Article.default;
  }
});
Object.defineProperty(exports, "Bodytext", {
  enumerable: true,
  get: function get() {
    return _Bodytext.default;
  }
});
Object.defineProperty(exports, "Branding", {
  enumerable: true,
  get: function get() {
    return _Branding.default;
  }
});
Object.defineProperty(exports, "Footer", {
  enumerable: true,
  get: function get() {
    return _Footer.default;
  }
});
Object.defineProperty(exports, "Heading", {
  enumerable: true,
  get: function get() {
    return _Heading.default;
  }
});
Object.defineProperty(exports, "Header", {
  enumerable: true,
  get: function get() {
    return _Header.default;
  }
});
Object.defineProperty(exports, "Hero", {
  enumerable: true,
  get: function get() {
    return _Hero.default;
  }
});
Object.defineProperty(exports, "Layout", {
  enumerable: true,
  get: function get() {
    return _Layout.default;
  }
});
Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function get() {
    return _Menu.default;
  }
});
Object.defineProperty(exports, "Page", {
  enumerable: true,
  get: function get() {
    return _Page.default;
  }
});
Object.defineProperty(exports, "Seo", {
  enumerable: true,
  get: function get() {
    return _Seo.default;
  }
});
Object.defineProperty(exports, "article", {
  enumerable: true,
  get: function get() {
    return _article.default;
  }
});
Object.defineProperty(exports, "bodytext", {
  enumerable: true,
  get: function get() {
    return _bodytext.default;
  }
});
Object.defineProperty(exports, "branding", {
  enumerable: true,
  get: function get() {
    return _branding.default;
  }
});
Object.defineProperty(exports, "footer", {
  enumerable: true,
  get: function get() {
    return _footer.default;
  }
});
Object.defineProperty(exports, "heading", {
  enumerable: true,
  get: function get() {
    return _heading.default;
  }
});
Object.defineProperty(exports, "header", {
  enumerable: true,
  get: function get() {
    return _header.default;
  }
});
Object.defineProperty(exports, "hero", {
  enumerable: true,
  get: function get() {
    return _hero.default;
  }
});
Object.defineProperty(exports, "layout", {
  enumerable: true,
  get: function get() {
    return _layout.default;
  }
});
Object.defineProperty(exports, "menu", {
  enumerable: true,
  get: function get() {
    return _menu.default;
  }
});
Object.defineProperty(exports, "page", {
  enumerable: true,
  get: function get() {
    return _page.default;
  }
});
Object.defineProperty(exports, "global", {
  enumerable: true,
  get: function get() {
    return _global.default;
  }
});

var _Article = _interopRequireDefault(require("./components/Article"));

var _Bodytext = _interopRequireDefault(require("./components/Bodytext"));

var _Branding = _interopRequireDefault(require("./components/Branding"));

var _Footer = _interopRequireDefault(require("./components/Footer"));

var _Heading = _interopRequireDefault(require("./components/Heading"));

var _Header = _interopRequireDefault(require("./components/Header"));

var _Hero = _interopRequireDefault(require("./components/Hero"));

var _Layout = _interopRequireDefault(require("./components/Layout"));

var _Menu = _interopRequireDefault(require("./components/Menu"));

var _Page = _interopRequireDefault(require("./components/Page"));

var _Seo = _interopRequireDefault(require("./components/Seo"));

var _article = _interopRequireDefault(require("./styles/article"));

var _bodytext = _interopRequireDefault(require("./styles/bodytext"));

var _branding = _interopRequireDefault(require("./styles/branding"));

var _footer = _interopRequireDefault(require("./styles/footer"));

var _heading = _interopRequireDefault(require("./styles/heading"));

var _header = _interopRequireDefault(require("./styles/header"));

var _hero = _interopRequireDefault(require("./styles/hero"));

var _layout = _interopRequireDefault(require("./styles/layout"));

var _menu = _interopRequireDefault(require("./styles/menu"));

var _page = _interopRequireDefault(require("./styles/page"));

var _global = _interopRequireDefault(require("./styles/global"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
